# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1422925097 18000
#      Mon Feb 02 19:58:17 2015 -0500
# Node ID 107ec5195a4735c081bcb57aeaac9261758e416c
# Parent  b95aab49b6beb88f87b6a2532e3cbafd70930f67
prepad.m: Fix str-to-num warnings (bug #44162)

* prepad.m: Use indexing rather than multiplication to expand constant value.
Fix adapted from cset c8e1b0213e34.  Add %!tests.
* postpad.m: Add %!tests.

diff --git a/scripts/general/postpad.m b/scripts/general/postpad.m
--- a/scripts/general/postpad.m
+++ b/scripts/general/postpad.m
@@ -89,12 +89,16 @@ endfunction
 %!assert (postpad ([1,2], 4), [1,2,0,0])
 %!assert (postpad ([1;2], 4), [1;2;0;0])
 %!assert (postpad ([1,2], 4, 2), [1,2,2,2])
 %!assert (postpad ([1;2], 4, 2), [1;2;2;2])
 %!assert (postpad ([1,2], 2, 2, 1), [1,2;2,2])
 %!assert (postpad ([1;2], 2, 2, 3), reshape ([1;2;2;2], 2, 1, 2))
 %!assert (postpad ([1,2], 2, 2, 3), reshape ([1,2,2,2], 1, 2, 2))
 
+%! ## Test with string concatenation (bug #44162)
+%!assert (postpad ("Octave", 16, "x"), "Octavexxxxxxxxxx")
+%!assert (postpad ("Octave", 4), "Octa")
+
 %!error postpad ()
 %!error postpad (1)
 %!error postpad (1,2,3,4,5)
 
diff --git a/scripts/general/prepad.m b/scripts/general/prepad.m
--- a/scripts/general/prepad.m
+++ b/scripts/general/prepad.m
@@ -75,33 +75,37 @@ function y = prepad (x, l, c, dim)
   d = sz (dim);
 
   if (d >= l)
     idx = repmat ({':'}, nd, 1);
     idx{dim} = d-l+1:d;
     y = x(idx{:});
   else
     sz (dim) = l - d;
-    y = cat (dim, c * ones (sz), x);
+    y = cat (dim, c(ones (sz)), x);
   endif
 
 endfunction
 
 
 %!assert (prepad ([1,2], 4), [0,0,1,2])
 %!assert (prepad ([1;2], 4), [0;0;1;2])
 
 %!assert (prepad ([1,2], 4, 2), [2,2,1,2])
 %!assert (prepad ([1;2], 4, 2), [2;2;1;2])
 
 %!assert (prepad ([1,2], 2, 2, 1), [2,2;1,2])
 
 %!assert (prepad ([1,2], 2, 2, 3), reshape ([2,2,1,2], 1, 2, 2))
 %!assert (prepad ([1;2], 2, 2, 3), reshape ([2;2;1;2], 2, 1, 2))
 
+%! ## Test with string concatenation (bug #44162)
+%!assert (prepad ("Octave", 16, "x"), "xxxxxxxxxxOctave")
+%!assert (prepad ("Octave", 4), "tave")
+
 ## FIXME: We need tests for multidimensional arrays.
 
 %!error prepad ()
 %!error prepad (1)
 %!error prepad (1,2,3,4,5)
 %!error <C must be empty or a scalar> prepad ([1,2], 2, ones (2))
 %!error <DIM must be an integer> prepad ([1,2], 2, 2, ones (3))
 %!error <DIM must be an integer> prepad ([1,2], 2, 2, 1.1)
